src/state/integrator.py

```diff
@@ -25,12 +24,34 @@ class IntegratorInformation:
     """A component of charm state that contains the configuration in integrator mode.
 
     Attributes:
-        backend_address: Configured backend ip address in integrator mode.
-        backend_port: Configured backend port in integrator mode.
+        backend_addresses: Configured list of backend ip addresses in integrator mode.
+        backend_ports: Configured list of backend ports in integrator mode.
     """
 
-    backend_address: IPvAnyAddress
-    backend_port: int = Field(gt=0, le=65535)
+    backend_addresses: list[IPvAnyAddress] = Field(
+        description="Configured list of backend ip addresses in integrator mode."
+    )
+    backend_ports: list[int] = Field(
```

Comment:
Something like `backend_ports: typing.Annotated[list[int], Field(gt=0, le=65535)]` should do the trick

Reply:
Ah thanks, didn't know that it works with a list.